pkg/controllers/openshift-controller-manager.go

```diff
@@ -129,8 +122,10 @@ func (s *OCPControllerManager) Run(ctx context.Context, ready chan<- struct{}, s
 		klog.Fatalf("failed to apply openshift namespaces %v", err)
 	}
 
-	options := openshift_controller_manager.OpenShiftControllerManager{Output: os.Stdout}
-	options.ConfigFilePath = s.ConfigFilePath
+	clientConfig, err := helpers.GetKubeClientConfig(configv1.KubeClientConfig{KubeConfig: s.kubeconfig})
+	if err != nil {
+		return err
+	}
 
-	return options.StartControllerManager(ctx.Done())
```

Comment:
How will the MicroShift service manager tell OCM to stop?

Reply:
With `StartControllerManager`, OCM should stop after it sees the context's done channel has closed. Under the proposed patch, `RunOpenShiftControllerManager` will ignore shutdown from the service manager and will keep running until the process dies.

Reply:
> With `StartControllerManager`, OCM should stop after it sees the context's done channel has closed. Under the proposed patch, `RunOpenShiftControllerManager` will ignore shutdown from the service manager and will keep running until the process dies.

Do you think we can pass the openshiftcontrolplanev1.OpenShiftControllerManagerConfig struct with all the configs to StartControllerManager ?

Reply:
I've created a go routing to pass the context as if the `StartControllerManager` function does.

Reply:
My assumption that StartControllerManager wires the provided channel to shutdown was wrong, so I don't think it necessary needs to be addressed here.

Reply:
it should be possible to gracefully shutdown once we get this in https://github.com/openshift/route-controller-manager/pull/6